packages/corejs/src/monitoring/index.tsx

```diff
@@ -0,0 +1,130 @@
+import * as Sentry from '@sentry/browser';
+import { BrowserTracing } from '@sentry/tracing';
+import Constants from 'expo-constants';
+import React, { createContext, useCallback, useContext, useEffect, useMemo } from 'react';
+import { Primitive } from '@sentry/types';
+
+import { MonitoringConfigType, MonitoringContextType, SentryTransactionStatus } from './types';
+
+export * from './types';
+
+/**
+ * Monitoring context which will create wrapper for monitoring functionality.
+*/
+export const MonitoringContext = createContext<MonitoringContextType | null>(null);
+
+/**
+ * Monitoring wrapper used to abstract Sentry functionality.
+ *
+ * @param {object} children - Provider children components that will be shared inside context provider.
+ * @param {MonitoringConfig} config - Configuration for sentry to override default configuration.
+*/
+export function MonitoringProvider({ children, config }: MonitoringConfigType) {
+  useEffect(() => {
+    Sentry.init({
+      dsn: (config.dsn || Constants.manifest.extra.SENTRY_DSN) as string,
+      environment: (config.environment || Constants.manifest.extra.ENV) as string,
+      debug: config.debug || Constants.manifest.extra.ENV !== 'production',
+      tracesSampleRate: config.tracesSampleRate || 0.5,
+      integrations: [
+        new BrowserTracing({ tracePropagationTargets: config.tracingOrigins }),
```

Comment:
please don't use Sentry constants in the corejs package. Only use the configuration object given as parameter. We can't use expo in corejs